usaspending_api/search/v2/views/spending_over_time.py

```diff
@@ -274,12 +358,50 @@ def build_elasticsearch_result(self, agg_response: AggResponse, time_periods: li
                         "Idv_Obligations": 0,
                         "Loan_Obligations": 0,
                         "Other_Obligations": 0,
+                        "total_outlays": None,
+                        "Contract_Outlays": None,
+                        "Direct_Outlays": None,
+                        "Grant_Outlays": None,
+                        "Idv_Outlays": None,
+                        "Loan_Outlays": None,
+                        "Other_Outlays": None,
                     }
                 )
 
         return results
 
-    def query_elasticsearch_for_prime_awards(self, time_periods: list) -> list:
+    def build_elasticsearch_result_awards(self, agg_response: AggResponse, time_periods: list) -> list:
+        """
+        In this function we are just taking the elasticsearch aggregate response and looping through the
+        buckets to create a results object for each time interval.
+
+        Using a min_date, max_date, and a frequency indicator generates either a list of dictionaries
+        containing fiscal year information (fiscal year, fiscal quarter, and fiscal month) or a list
+        of dictionaries containing calendar year information (calendar year). The following are the format
+        of date_range based on the frequency:
+            * "calendar_year" returns a list of dictionaries containing {calendar year}
+            * "fiscal_year" returns list of dictionaries containing {fiscal year}
+            * "quarter" returns a list of dictionaries containing {fiscal year and quarter}
+            * "month" returns a list of dictionaries containg {fiscal year and month}
+        NOTE the generate_date_range() can also generate non fiscal date range (calendar ranges) as well.
+        """
+
+        results = []
+        date_buckets = agg_response.group_by_time_period.buckets
+        parsed_bucket = None
+
+        if date_buckets is not None:
+            for bucket in date_buckets:
+                parsed_bucket = self.parse_elasticsearch_bucket(bucket.to_dict())
+
+                if parsed_bucket is not None:
```

Comment:
Is this conditional necessary? I see you set it to `None` above, but won't `parse_elasticsearch_bucket` always return a dictionary?

Reply:
Good catch. This was left over from the original, Transaction, code where `parsed_bucket` can be `None`. I've removed this check.